sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -413,6 +413,7 @@ object Decimal {
   val ROUND_HALF_EVEN = BigDecimal.RoundingMode.HALF_EVEN
   val ROUND_CEILING = BigDecimal.RoundingMode.CEILING
   val ROUND_FLOOR = BigDecimal.RoundingMode.FLOOR
+  val ROUND_DOWN = BigDecimal.RoundingMode.DOWN
```

Comment:
Need this change, otherwise:
```
Caused by: org.codehaus.commons.compiler.CompileException: File 'generated.java', Line 41, Column 138: failed to compile: org.codehaus.commons.compiler.CompileException: File 'generated.java', Line 41, Column 138: A method named "ROUND_DOWN" is not declared in any enclosing class nor any supertype, nor through a static import
```

Reply:
I guess we need to modify `Decimal.changePrecision()` as well? Could you add it to [DecimalSuite.scala#L207](https://github.com/apache/spark/blob/479b31fa046e8402f4f93cdbad5fe93ef1ea570f/sql/catalyst/src/test/scala/org/apache/spark/sql/types/DecimalSuite.scala#L207) to check?

Reply:
I added it and seem do not need change `longVal`. This test case can passed.